File: medium/39. cards/cards.c
#include <stdio.h>

void find_missing_cards(int N, long long S, long long S_square, long long S_cube) {
    long long S_total = (long long)N * (N + 1) / 2;
    long long S_square_total = (long long)N * (N + 1) * (2 * N + 1) / 6;
    long long S_cube_total = S_total * S_total;

    long long d1 = S_total - S;
    long long d2 = S_square_total - S_square;
    long long d3 = S_cube_total - S_cube;

    for (int x = 1; x <= N; ++x) {
        for (int y = x + 1; y <= N; ++y) {
            int z = d1 - (x + y);
            if (z <= y || z > N) continue;
            if ((long long)x * x + (long long)y * y + (long long)z * z == d2 &&
                (long long)x * x * x + (long long)y * y * y + (long long)z * z * z == d3) {
                printf("%d %d %d\n", x, y, z);
                return;
                }
        }
    }
}

int main() {
    int N;
    long long S1, S2, S3;

    scanf("%d", &N);
    scanf("%lld %lld %lld", &S1, &S2, &S3);

    find_missing_cards(N, S1, S2, S3);

    return 0;
}

File: medium/39. cards/cards.py
import sys

def find_missing_cards(N, S, S_square, S_cube):
    S_total = N * (N+1) / 2
    S_square_total = N * (N+1) * (2*N+1) / 6
    S_cube_total = (N*(N+1)/2) * (N*(N+1)/2)
    d1 = S_total - S
    d2 = S_square_total - S_square
    d3 = S_cube_total - S_cube

    for x in range(0, N+1):
        for y in range(x + 1, N+1):
            z = d1 - x - y
            if z > y and z <= N and x ** 2 + y ** 2 + z ** 2 == d2 and x ** 3 + y ** 3 + z ** 3 == d3:
                    return int(x), int(y), int(z)

def main():
    N = int(input())
    S, S_square, S_cube = map(int, input().split())
    print(*find_missing_cards(N, S, S_square, S_cube))

if __name__ == '__main__':
	main()
